magma/tuple.py

```diff
@@ -585,23 +599,25 @@ def is_wireable(cls, rhs):
         rhs = magma_type(rhs)
         if (
             not isinstance(rhs, AnonProductKind) or
-            len(cls.fields) != len(rhs.fields)
+            list(cls.field_dict.keys()) != list(rhs.field_dict.keys())
         ):
             return False
         for k, v in cls.field_dict.items():
-            if k not in rhs.field_dict or not v.is_wireable(rhs.field_dict[k]):
+            v = magma_type(v)
```

Comment:
it seems like we are doubly calling magma_type here and inside is_wireable? the interface to that function should be clear whether it accepts a type or (type | magma_protocol). and in that case just pass in the raw "type" instead of calling magma_type out here.